src/Autocomplete.tsx

```diff
@@ -0,0 +1,83 @@
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { Person } from './types/Person';
+
+interface Props {
+  people: Person[]
+  delay?: number
+  onSelected: (value: Person | null) => void
+}
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  delay,
+  onSelected,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [delay]);
+
+  const filteredPeople = useMemo(() => {
+    const normalizedQuery = appliedQuery.toLowerCase().trim();
+
+    return people.filter(
+      person => person.name.toLowerCase().includes(normalizedQuery),
+    );
+  }, [appliedQuery, people]);
+
+  const handleSelect = (person: Person) => {
+    setQuery(person.name);
+    onSelected(person);
+    setIsDropdownOpen(false);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    onSelected(null);
+  };
+
+  return (
+    <>
+      <div className={cn('dropdown', {
+        'is-active': isDropdownOpen,
+      })}
+      >
+        <div className="dropdown-trigger">
+          <input
+            type="text"
+            placeholder="Enter a part of the name"
+            className="input"
+            value={query}
+            onFocus={() => setIsDropdownOpen(true)}
+            onBlur={() => setIsDropdownOpen(false)}
+            onChange={handleInputChange}
+          />
+        </div>
+
+        <div className="dropdown-menu" role="menu">
+          {filteredPeople.length >= 1 ? (
+            <div className="dropdown-content">
+              {filteredPeople.map(person => (
+                <a
+                  className="dropdown-item"
+                  href="/"
+                  key={person.slug}
+                  onMouseDown={() => handleSelect(person)}
+                >
+                  <p className="has-text-link">{person.name}</p>
+                </a>
+              ))}
+            </div>
+          ) : (
+            <p className="help is-danger">No matching suggestions</p>
```

Comment:
I'd show it inside the popup - and not as an error, like one of suggestions, with black text

but it's not critical, and there are no such requirements in the task, so up to you